### ISSUE TITLE
fetching data/xx interacts poorly with any additional dependencies

### ISSUE THREAD

**Issue by aappling-usgs** [USGS-VIZLAB/vizlab#207]
There's a scenario where having all fetch items depend on their respective config files is a problem: when the fetch item is a file in data and therefore never gets copied into the cache, its timestamp never changes. So if it depends on cache/config/xx.rds, it gets 'fetched' every time. The fetch is just a few shell & function calls, no real action, but it still adds to every build time by a second or two. My example is iris-data in USGS-VIZLAB/example; it may be less common in real vizzies, but it didn't take long to find this real example: https://github.com/USGS-VIZLAB/water-use/blob/master/viz.yaml#L121.

It could be that we just need to remove the config/xx.rds dependency for items already in data/. or maybe we `touch` the data file after a change to the config, or `touch` the config to appear older than the data file? Or maybe we require & check that the config doesn't contain any parameters (because data can't use them) and _then_ remove the config/xx.rds dependency? Or maybe we copy data/ files into cache/ and don't permit targets in the data/ folder? I don't see a clear winner yet.

This issue emerged during https://github.com/USGS-VIZLAB/vizlab/issues/193 but could be addressed independently of it, depending on the solution we choose.

**Comment by aappling-usgs** [comment]
possible solutions
* status file to represent data files
OR
* have fetch.file actually update the timestamp of data/iris.csv

laura points out that if we don't fix this this sprint, the effect is to allow builds to take longer than they would otherwise. could be worse.

**Comment by jread-usgs** [comment]
not sure if I am not interpreting this correctly, but we also have a common fetch pattern that gets re-run each time when we use packaged geospatial data from the `maps` package. See irma or harvey's use of counties, states, and islands. I think these `Spatial` objects are re-built every time we run fetch even though the underlying package data used to create them does not change (i.e., the version of `maps` or `mapdata` packages). Not a big deal, but this does add a few seconds per target and may be bigger for other vizzies (maybe all US counties...)

**Comment by aappling-usgs** [comment]
@jread-usgs - i think that particular fetch pattern can be explained by #192 and does not need to be re-addressed here, though specific confirmation that your use case has been resolved (in that issue) would be nice.

**Comment by aappling-usgs** [comment]
I think this will not be a problem with remake. `mayfly_nymph` in the testviz is an example, because the data file doesn't get updated or copied into cache:
```yml
  -
    id: mayfly_nymph
    location: data/mayfly_nymph.csv
    fetcher: mayfly_nymph
    info: insect
    scripts: scripts/fetch/mayfly_nymph.R
    reader: my_csv_reader
    mimetype: text/csv
    export: true
```
```r
fetch.mayfly_nymph <- function(viz) {
  # this function doesn't actually do, or need to do, anything
  message("fetching mayfly_nymph from ", viz[['location']], " to ", viz[['location']])
  message("a mayfly is a kind of ", viz$info)
}
```

and we have a test in tests-timestamps.R to confirm that it only gets built the first time:
```r
test_that("alwaysCurrent doesn't get rebuilt unless missing", {
  # confirm that mayfly_nymph is an alwaysCurrent item
  source('scripts/fetch/mayfly_nymph.R')
  expect_equal(fetchTimestamp.mayfly_nymph, alwaysCurrent)
  
  # first time: fetches
  suppressWarnings(expect_message(vizmake('mayfly_nymph'), '[ BUILD ] data/mayfly_nymph.csv', fixed=TRUE))
  
  # after that: doesn't fetch
  suppressWarnings(expect_message(vizmake('mayfly_nymph'), '[ ----- ] mayfly_nymph', fixed=TRUE))
})
```

I will wait until we have argument dependencies (#287) fully implemented to officially close this issue, but I think switching to remake is our answer.

**Comment by aappling-usgs** [comment]
shown to be resolved in #321
